File: ALODWebUtility/Printing/PDFString.vb
Namespace Printing

    Public Class PDFString
        Private _alignment As String = "center"
        Private _color As String = "#FF0000/90"
        Private _font As String = "arial"
        Private _fontSize As Integer = 12
        Private _fontWeight As String = String.Empty
        Private _linespacing As Integer = 0
        Private _postNewLines As Integer = 0
        Private _preNewLines As Integer = 0
        Private _rotation As Integer = 0
        Private _text As String = String.Empty

        Public Property Alignment() As String
            Get
                Return Me._alignment
            End Get
            Set(value As String)
                Me._alignment = value
            End Set
        End Property

        Public Property Color() As String
            Get
                Return Me._color
            End Get
            Set(value As String)
                Me._color = value
            End Set
        End Property

        Public Property Font() As String
            Get
                Return Me._font
            End Get
            Set(value As String)
                Me._font = value
            End Set
        End Property

        Public Property FontSize As Integer
            Get
                Return Me._fontSize
            End Get
            Set(value As Integer)
                Me._fontSize = value
            End Set
        End Property

        Public Property FontWeight() As String
            Get
                Return Me._fontWeight
            End Get
            Set(value As String)
                Me._fontWeight = value
            End Set
        End Property

        Public Property Linespacing As Integer
            Get
                Return Me._linespacing
            End Get
            Set(value As Integer)
                Me._linespacing = value
            End Set
        End Property

        Public Property PostNewLines As Integer
            Get
                Return Me._postNewLines
            End Get
            Set(value As Integer)
                Me._postNewLines = value
            End Set
        End Property

        Public Property PreNewLines As Integer
            Get
                Return Me._preNewLines
            End Get
            Set(value As Integer)
                Me._preNewLines = value
            End Set
        End Property

        Public Property Rotation As Integer
            Get
                Return Me._rotation
            End Get
            Set(value As Integer)
                Me._rotation = value
            End Set
        End Property

        Public Property Text() As String
            Get
                Return Me._text
            End Get
            Set(value As String)
                Me._text = value
            End Set
        End Property

        Public Function GetHTML() As String
            Dim htmlString As String = String.Empty

            Dim sLinespacing As String = String.Empty
            Dim sSize As String = String.Empty
            Dim sAlignment As String = String.Empty
            Dim sFontFace As String = String.Empty
            Dim sColor As String = String.Empty

            If (Linespacing <> 0) Then
                sLinespacing = "linespacing=" & Linespacing.ToString()
            End If

            If (FontSize >= 0) Then
                sSize = "fontsize=" & FontSize.ToString()
            End If

            If (Not String.IsNullOrEmpty(Alignment)) Then
                sAlignment = "align=" & Alignment
            End If

            If (Not String.IsNullOrEmpty(Font)) Then
                sFontFace = "face='" & Font
            End If

            If (Not String.IsNullOrEmpty(FontWeight)) Then
                sFontFace = sFontFace & "-" & FontWeight & "'"
            Else
                sFontFace = sFontFace & "'"
            End If

            If (Not String.IsNullOrEmpty(Color)) Then
                sColor = "color=" & Color
            End If

            htmlString = GetBreakTagString(PreNewLines)
            htmlString = htmlString & "<p " & Linespacing & " " & sSize & " " & sAlignment + "><StyleRun rotate=" & Rotation & "><Font " & sFontFace & " " & sColor & ">" & Text & "</Font></StyleRun></p>"
            htmlString = htmlString & GetBreakTagString(PostNewLines)

            Return htmlString
        End Function

        Private Function GetBreakTagString(ByVal numberOfBreaks As Integer) As String
            If (numberOfBreaks <= 0) Then
                Return String.Empty
            End If

            Dim breakTags As String = String.Empty

            For i As Integer = 1 To numberOfBreaks
                breakTags = breakTags & "<br>"
            Next

            Return breakTags
        End Function

    End Class

End Namespace

File: ALODWebUtility/Printing/PDFString.cs
namespace ALODWebUtility.Printing
{
    public class PDFString
    {
        private string _alignment = "center";
        private string _color = "#FF0000/90";
        private string _font = "arial";
        private int _fontSize = 12;
        private string _fontWeight = string.Empty;
        private int _linespacing = 0;
        private int _postNewLines = 0;
        private int _preNewLines = 0;
        private int _rotation = 0;
        private string _text = string.Empty;

        public string Alignment
        {
            get { return this._alignment; }
            set { this._alignment = value; }
        }

        public string Color
        {
            get { return this._color; }
            set { this._color = value; }
        }

        public string Font
        {
            get { return this._font; }
            set { this._font = value; }
        }

        public int FontSize
        {
            get { return this._fontSize; }
            set { this._fontSize = value; }
        }

        public string FontWeight
        {
            get { return this._fontWeight; }
            set { this._fontWeight = value; }
        }

        public int Linespacing
        {
            get { return this._linespacing; }
            set { this._linespacing = value; }
        }

        public int PostNewLines
        {
            get { return this._postNewLines; }
            set { this._postNewLines = value; }
        }

        public int PreNewLines
        {
            get { return this._preNewLines; }
            set { this._preNewLines = value; }
        }

        public int Rotation
        {
            get { return this._rotation; }
            set { this._rotation = value; }
        }

        public string Text
        {
            get { return this._text; }
            set { this._text = value; }
        }

        public string GetHTML()
        {
            string htmlString = string.Empty;

            string sLinespacing = string.Empty;
            string sSize = string.Empty;
            string sAlignment = string.Empty;
            string sFontFace = string.Empty;
            string sColor = string.Empty;

            if (Linespacing != 0)
            {
                sLinespacing = "linespacing=" + Linespacing.ToString();
            }

            if (FontSize >= 0)
            {
                sSize = "fontsize=" + FontSize.ToString();
            }

            if (!string.IsNullOrEmpty(Alignment))
            {
                sAlignment = "align=" + Alignment;
            }

            if (!string.IsNullOrEmpty(Font))
            {
                sFontFace = "face='" + Font;
            }

            if (!string.IsNullOrEmpty(FontWeight))
            {
                sFontFace = sFontFace + "-" + FontWeight + "'";
            }
            else
            {
                sFontFace = sFontFace + "'";
            }

            if (!string.IsNullOrEmpty(Color))
            {
                sColor = "color=" + Color;
            }

            htmlString = GetBreakTagString(PreNewLines);
            htmlString = htmlString + "<p " + Linespacing + " " + sSize + " " + sAlignment + "><StyleRun rotate=" + Rotation + "><Font " + sFontFace + " " + sColor + ">" + Text + "</Font></StyleRun></p>";
            htmlString = htmlString + GetBreakTagString(PostNewLines);

            return htmlString;
        }

        private string GetBreakTagString(int numberOfBreaks)
        {
            if (numberOfBreaks <= 0)
            {
                return string.Empty;
            }

            string breakTags = string.Empty;

            for (int i = 1; i <= numberOfBreaks; i++)
            {
                breakTags = breakTags + "<br>";
            }

            return breakTags;
        }
    }
}
